airflow/serialization/schema.json

```
@@ -90,6 +90,7 @@
             { "$ref": "#/definitions/typed_relativedelta" }
           ]
         },
+        "schedule_on": { "type": "array" },
```

Could you create a dataset def (dict with uri and extra properties) and then 
```suggestion
        "schedule_on": {
            "type": "array",
            "items": { "$ref": "#/definitions/dataset" }
        },
```

---

yup 👍